lib/que/adapters/active_record_with_lock.rb

```diff
@@ -35,13 +47,22 @@ def execute(command, params = [])
       def lock_job_with_lock_database(queue, cursor)
         result = []
         loop do
-          result = Que.execute(:find_job_to_lock, [queue, cursor])
+          result = Que.transaction do
+            observe(nil, FindJobSecondsTotal, queue: queue) do
+              result = Que.execute(:find_job_to_lock, [queue, cursor])
+            end
 
-          break if result.empty?
+            return result if result.empty?
 
-          cursor = result.first["job_id"]
-          break if pg_try_advisory_lock?(cursor)
+            cursor = result.first["job_id"]
+            job_locked = pg_try_advisory_lock?(cursor)
+
+            observe(FindJobHitTotal, nil, { queue: queue, job_hit: job_locked })
+            return result if job_locked
+          end
+          break if result
```

Comment:
Nit: a combination of returns and breaks can be a bit hard to follow. Is it possible to avoid this break such that all result cases are handled by early returns?